stream_alert/rule_processor/handler.py

```diff
@@ -112,17 +120,23 @@ def _process_alerts(self, classifier, payload, data):
         """
         classifier.classify_record(payload, data)
         if not payload.valid:
-            LOGGER.error('Invalid data: %s\n%s', payload, data)
+            # Log a message about this failure if this is not a development env
+            if self.env['lambda_alias'] != 'development':
+                LOGGER.error('Log failed to match any defined schemas: %s\n%s', payload, data)
+            self._failed_log_count += 1
             return
 
         alerts = StreamRules.process(payload)
         if not alerts:
-            LOGGER.debug('Valid data, no alerts')
+            pluralize = 's' if len(payload.records) > 1 else ''
```

Comment:
luckily `alerts` should never be of `NoneType` here so I can move this outside of the conditional without a problem